c++/src/capnp/compiler/capnpc-c++.c++

```diff
@@ -1785,15 +1785,15 @@ private:
             "      ::capnp::bounded<", offset, ">() * ::capnp::POINTERS), kj::mv(value));\n"
             "}\n",
             COND(type.hasDisambiguatedTemplate(),
-                "#ifndef _MSC_VER\n"
+                "#if !defined(_MSC_VER) || defined(__clang__)\n"
```

Comment:
Since this touches code generation, you'll want to run `./regenerate-bootstraps.sh` in capnproto/c++, then check in the regenerated code in a second commit.

Reply:
Hmm we have a unit test that should have failed if the bootstraps weren't regenerated. Is it possible this line doesn't appear in any of the bootstraps?

Reply:
I did also change all the generated code by hand. However, I did notice that when I reran `regenerate-bootstrap.sh`  I got a bunch of changes so maybe there's a bug in that somewhere?

```
-  ::capnp::Capability::Server::DispatchCallResult dispatchCall(
-      uint64_t interfaceId, uint16_t methodId,
+  ::kj::Promise<void> dispatchCall(uint64_t interfaceId, uint16_t methodId,
```

This is in addition to all the CAPNP_VERSION checks that switch from `!= 7000` to `!= 9000`